packages/svelte/src/internal/client/operations.js

```diff
@@ -1,46 +1,115 @@
-
-// We cache the Node and Element prototype methods, so that subsequent calls-sites are monomorphic rather
-// than megamorphic.
-const node_prototype = /** @type {Node} */ (has_browser_globals ? Node.prototype : {});
-const element_prototype = /** @type {Element} */ (has_browser_globals ? Element.prototype : {});
-const text_prototype = /** @type {Text} */ (has_browser_globals ? Text.prototype : {});
-const map_prototype = Map.prototype;
-const append_child_method = node_prototype.appendChild;
-const clone_node_method = node_prototype.cloneNode;
-const map_set_method = map_prototype.set;
-const map_get_method = map_prototype.get;
-const map_delete_method = map_prototype.delete;
-// @ts-expect-error improve perf of expando on DOM events
-element_prototype.__click = undefined;
-// @ts-expect-error improve perf of expando on DOM text updates
-text_prototype.__nodeValue = ' ';
-// @ts-expect-error improve perf of expando on DOM className updates
-element_prototype.__className = '';
-
-const first_child_get = /** @type {(this: Node) => ChildNode | null} */ (
-	// @ts-ignore
-	has_browser_globals ? get_descriptor(node_prototype, 'firstChild').get : null
-);
-
-const next_sibling_get = /** @type {(this: Node) => ChildNode | null} */ (
-	// @ts-ignore
-	has_browser_globals ? get_descriptor(node_prototype, 'nextSibling').get : null
-);
-
-const text_content_set = /** @type {(this: Node, text: string ) => void} */ (
-	// @ts-ignore
-	has_browser_globals ? get_descriptor(node_prototype, 'textContent').set : null
-);
-
-const class_name_set = /** @type {(this: Element, class_name: string) => void} */ (
-	// @ts-ignore
-	has_browser_globals ? get_descriptor(element_prototype, 'className').set : null
-);
+// We cache the Node and Element prototype methods, so that we can avoid doing
+// expensive prototype chain lookups.
+
+/**
+ * @type {Node}
+ */
+var node_prototype;
+/**
+ * @type {Element}
+ */
+var element_prototype;
+/**
+ * @type {Text}
+ */
+var text_prototype;
+/**
+ * @type {Map<any, any>}
+ */
+var map_prototype;
+/**
+ * @type {typeof Node.prototype.appendChild}
+ */
+var append_child_method;
+/**
+ * @type {typeof Node.prototype.cloneNode}
+ */
+var clone_node_method;
+/**
+ * @type {typeof Map.prototype.set}
+ */
+var map_set_method;
+/**
+ * @type {typeof Map.prototype.get}
+ */
+var map_get_method;
+/**
+ * @type {typeof Map.prototype.delete}
+ */
+var map_delete_method;
+/**
+ * @type {(this: Node) => ChildNode | null}
+ */
+var first_child_get;
+/**
+ * @type {(this: Node) => ChildNode | null}
+ */
+var next_sibling_get;
+/**
+ * @type {(this: Node, text: string ) => void}
+ */
+var text_content_set;
+
+/**
+ * @type {(this: Element, class_name: string) => void}
+ */
+var class_name_set;
+
+// export these for reference in the compiled code, making global name deduplication unnecessary
+/**
+ * @type {Window}
+ */
+export var $window;
+/**
+ * @type {Document}
+ */
+export var $document;
+
+export function init_operations() {
```

Comment:
```suggestion
/**
 * Initialize these lazily to avoid issues when using the runtime in a server context
 * where these globals are not available while avoiding a separate server entry point
 */
export function init_operations() {
```